api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1757,8 +1757,11 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
                     "{\"inline\": {\"no_encrypt\": True}}"
                 )
 
-        logger.info("Refresh enclosure id on the system")
+        logger.info(
+             "Encrypt pillar values and Refresh enclosure id on the system"
+        )
         for state in [
+            'components.system.config.pillar_encrypt',
```

Comment:
How does just encrypting the pillar solves this issue?

Reply:
I should have mentioned, my bad!
In `pillar_encrypt`, we install cortx module, which was expected for utils (used by `lyveutils` in `controller_cli`). Since that was missing, `enclosure-id` file was not getting written with the expected value and iterating over 0.